terminus/builders/procedural_city/vertex_graph_to_roads_converter.py

```diff
@@ -49,12 +55,35 @@ def _build_road(self, street, previous_vertex, current_vertex):
                 edges.append([neighbour,
                              abs(current_angle - neighbours_angle)])
             edges.sort(key=itemgetter(1))
-            if edges and (edges[0][1] <= self.angle_threshold):
-                selected_neighbour = edges[0][0]
-                current_vertex.neighbours.remove(selected_neighbour)
-                selected_neighbour.neighbours.remove(current_vertex)
-                street.add_point(selected_neighbour.coords)
-                self._build_road(street, current_vertex, selected_neighbour)
+            # Pick the edge with the lowest angle and check if it's below the
```

Comment:
Can you factor this out? This method ended up being quite large and this piece of code has a well defined semantic (picking the best neighbor)